vulkan_generator/vulkan_parser/types.py

```diff
@@ -262,6 +262,54 @@ class AllVulkanCommands:
     command_aliases: Dict[str, VulkanCommandAlias] = field(default_factory=dict)
 
 
+@dataclass
+class VulkanFeatureExtension:
+    """The base structure to extend a Vulkan feature"""
+
+
+@dataclass
+class VulkanFeatureExtensionEnum(VulkanFeatureExtension):
+    """
+    The structure to hold extension to an existing enum
+
+    When a vulkan enum is a required feature, unlike other types, the extending fields are defined
+    on the feature rather than the original enum
+    """
+    # Which type this enum is extending
+    basetype: str
+
+    # Extending enum field is an alias to another enum fields
+    alias: Optional[str]
+
+    # Extension number is used when calculating enum field value
+    extnumber: Optional[str]
+
+    # Offset is used when calculating enum field value
+    offset: Optional[str]
+
+    # Bitpos is used when calcualting bitfield value
+    bitpos: Optional[str]
+
+    # Some features explicitly gives the new value
+    value: Optional[str]
+
+
+@dataclass
+class VulkanFeature:
+    """The structure to hold a required feature for a Vulkan Core Version"""
+    name: str
+    feature_type: str
+    feature_extension: Optional[VulkanFeatureExtension]
+
+
+@dataclass
+class VulkanCoreVersion:
+    """The structure to hold all the features required or added by a Vulkan core version"""
+    name: str
+    number: str
```

Comment:
Feels a bit weird that we're treating version numbers as strings everywhere, but I guess this is easier and we never know if they'll make a VulkanES1.1 or something in the future

Reply:
The issue is that number is "1.0", "1.1" go on. So as they are actually "not numbers" and it would be weird to store them in a floating point number, the only option was store them in a string.